crates/storage/src/table/read/array.rs

```diff
@@ -170,6 +171,33 @@ fn read_binary(
 }
 
 
+fn read_fixed_size_binary(
+    storage: &impl Storage,
+    pos: usize,
+    ranges: Option<&RangeList<u32>>,
+    size: i32
+) -> anyhow::Result<ArrayRef>
+{
+    let nulls = storage.read_null_mask(pos, ranges)
+        .context("failed to read null mask")?;
+
+    let value_ranges = ranges.map(|list| {
+        list.iter()
+            .map(|r| (r.start * size as u32)..(r.end * size as u32))
+            .collect::<Vec<_>>()
+            .try_into()
```

Comment:
Let's use [RangeList::new_unchecked()](https://github.com/subsquid/data/blob/4466c34868f24b0510efad778f774608dc56816f/crates/primitives/src/range/mod.rs#L44).

P.S.

This is another point where materialisation is not necessary. Eliminating it can make a difference, but let's leave this for future work.

Reply:
Why is this `RangeList` another type instead of being `impl index::RangeList`?

Reply:
Here `RangeList` comes with some guarantees:

* ranges are ordered
* ranges never intersect

While `sqd_array::index::RangeList` is an arbitrary list of ranges.